src/foam/dao/daoUtils.js

```diff
@@ -265,6 +265,56 @@ foam.CLASS({
 });
 
 
+foam.CLASS({
+  package: 'foam.dao',
+  name: 'ObjectMutationProxyDAO',
+  extends: 'foam.dao.ProxyDAO',
+  flags: ['java'],
+  abstract: true,
+
+  javaImports: [
+    'foam.core.Detachable',
+    'foam.core.FObject',
+    'foam.core.X',
+    'foam.dao.ProxySink'
+  ],
+
+  axioms: [
+    {
+      name: 'javaExtras',
+      buildJavaClass: function(cls) {
+        cls.extras.push(`
+          public abstract FObject update(X x, FObject obj);
+        `);
+      }
+    }
+  ],
+
+  methods: [
+    {
+      name: 'find_',
+      javaCode: `
+        return update(
+          x, getDelegate().find_(x, id));
+      `
+    },
+    {
+      name: 'select_',
+      javaCode: `
+        if ( sink == null ) sink = new ArraySink();
+        ProxySink refinedSink = new ProxySink(x, sink) {
+          @Override
+          public void put(Object obj, Detachable sub) {
+            super.put(update(x, (FObject) obj), sub);
+          }
+        };
+        return ((ProxySink) super.select_(x, refinedSink, skip, limit, order, predicate)).getDelegate();
+      `
```

Comment:
update on select_ and find_? should be used sparingly.

Reply:
Yes this is true. This was going to be used for an `isVisible` storageTransient property it if wasn't possible to do without it. As it happens it was, and this is no longer being used, however we do have DAO decorators on the system which do this which could extend this class for simplicity, so I left it in the PR.

Reply:
This has been removed